src/Api/Request.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Drupal\commerce_canadapost\Api;
+
+use Drupal\commerce_canadapost\Plugin\Commerce\ShippingMethod\CanadaPost;
+use Drupal\commerce_store\Entity\StoreInterface;
+
+use Exception;
+
+/**
+ * CanadaPost API Service.
+ *
+ * @package Drupal\commerce_canadapost
+ */
+abstract class Request implements RequestInterface {
+
+  /**
+   * The logger channel factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $logger;
+
+  /**
+   * Request class constructor.
+   */
+  public function __construct() {
+    $this->logger = \Drupal::service('logger.factory')->get(COMMERCE_CANADAPOST_LOGGER_CHANNEL);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getApiSettings(StoreInterface $store = NULL, CanadaPost $shipping_method = NULL) {
+    $api_settings = [];
+
+    if (!$store && !$shipping_method) {
+      throw new Exception('A shipping method or a store is required to fetch the Canada Post API settings.');
+    }
+
+    // Check if we have settings set on the shipping method, if so, use that.
+    if ($shipping_method && $shipping_method->apiIsConfigured()) {
+      $api_settings = $shipping_method->getConfiguration()['api'];
+    }
+    // Else, we fallback to the store API settings.
+    elseif ($store) {
+      $api_settings = $this->decodeSettings(
+        $store->get('canadapost_api_settings')->getValue()[0]['value']
+      );
+    }
```

Comment:
**CHANGE REQUEST**

Keep in mind that based on the changes suggested the store settings might be empty, so you'd need to check here if the store exists + if it has settings, like you do for the shipping method.

Reply:
As mentioned above ^^, there shouldn't be a case with a store w/ empty settings because the API settings are required in the store form.

Reply:
Never mind. I didn't read the comment about the store settings being optional until further down. We're making sure the store has settings now.